syncer/syncer.go

```diff
@@ -2050,22 +2045,17 @@ func (s *Syncer) reopenWithRetry(cfg replication.BinlogSyncerConfig) error {
 }
 
 func (s *Syncer) reopen(cfg replication.BinlogSyncerConfig) (streamer.Streamer, error) {
-	if s.syncer != nil {
-		err := s.closeBinlogSyncer(s.syncer)
-		s.syncer = nil
+	if s.streamerProducer != nil {
```

Comment:
should we check whether it's `remote binlog reader`?

Reply:
only remote binlog reader call this function currently, but adding a check is better